gitlab/_backends/requests_backend.py

```diff
@@ -64,18 +75,18 @@ def prepare_send_data(
                 if TYPE_CHECKING:
                     assert isinstance(post_data, dict)
                 for k, v in post_data.items():
-                    if isinstance(v, bool):
+                    if isinstance(v, (bool, int)):
```

Comment:
This is the real fix here, but maybe we could already future-proof it a bit, WDYT?

```suggestion
                    if isinstance(v, (bool, complex, float, int)):
```

So we don't get similar issues in the future. These should probably be enough according to https://en.wikipedia.org/wiki/Python_(programming_language)#/media/File:Python_3._The_standard_type_hierarchy-en.svg

Reply:
But below we convert it to an int and then a str... And the only time we go through this path is when there is a file.

So I am unsure. What do you think?

Reply:
@nejch I did what you suggested. Well a bit more than you suggested 🙂

Reply:
Ah yes, makes sense like that, missed it. TBH I checked https://github.com/python-gitlab/python-gitlab/pull/1404 and don't see why we should really be converting the boolean to int, but maybe we can keep it for now.

Yeah, seems like it can happen whenever someone adds a file-type attribute but then affects other attributes as we switch to the multipart encoder for everything.